pandas/_libs/testing.pyx

```diff
@@ -119,7 +121,7 @@ cpdef assert_almost_equal(a, b,
             f"Can't compare objects without length, one or both is invalid: ({a}, {b})"
         )
 
-        if a_is_ndarray and b_is_ndarray:
+        if (a_is_ndarray and b_is_ndarray) or (a_has_size_and_shape and b_has_size_and_shape):
```

Comment:
arent the is_array checks redundant here?

Reply:
Probably, I had kept them in case there was some edge case I hadn't thought of but I can also remove them